app/com/linkedin/drelephant/tuning/Constant.java

```diff
@@ -15,4 +11,44 @@
   public enum ExecutionEngineTypes{MR,SPARK}
   public enum TypeofManagers{AbstractBaselineManager,AbstractFitnessManager,AbstractJobStatusManager,AbstractParameterGenerateManager}
 
+
+  public static final String JVM_MAX_HEAP_MEMORY_REGEX = ".*-Xmx([\\d]+)([mMgG]).*";
+  public static final double YARN_VMEM_TO_PMEM_RATIO = 2.1;
+  public static final int MB_IN_ONE_GB = 1024;
+  public static final int SORT_BUFFER_CUSHION = 769;
+  public static final int DEFAULT_CONTAINER_HEAP_MEMORY = 1536;
+  public static final int OPTIMAL_MAPPER_SPEED_BYTES_PER_SECOND = 10;
+  public static final double HEAP_MEMORY_TO_PHYSICAL_MEMORY_RATIO = 0.75D;
+  public static final double SORT_SPILL_PERCENTAGE_THRESHOLD = 0.85D;
+  public static final double MAPPER_MEMORY_SPILL_THRESHOLD_1 = 2.7D;
+  public static final double MAPPER_MEMORY_SPILL_THRESHOLD_2 = 2.2D;
+  public static final double MEMORY_TO_SORT_BUFFER_RATIO = 1.6;
+  public static final double SPILL_PERCENTAGE_STEP_SIZE = 0.05d;
+
+  public enum HeuristicsToTuneByPigHbt {
+    MAPPER_MEMORY("Mapper Memory"),
+    MAPPER_TIME("Mapper Time"),
+    MAPPER_SPILL("Mapper Spill"),
+    MAPPER_SPEED("Mapper Speed"),
+    REDUCER_MEMORY("Reducer Memory"),
+    REDUCER_TIME("Reducer Time");
```

Comment:
Reducer time will be fixed by setting appropriate #reducers, but since we cannot set those as of now so I haven't introduced the logic for the same. As soon as we get this resolved after discussion with the respective team I would append the appropriate logic.

Reply:
Make sense . Please add todo , so that we can take in future

Reply:
Done